seqr/views/apis/staff_api_tests.py

```diff
@@ -531,3 +608,82 @@ def test_anvil_export(self, mock_export_multiple_files):
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['projectsByGuid', 'locusListsByGuid', 'savedVariantsByGuid', 'variantFunctionalDataByGuid', 'genesById', 'variantNotesByGuid', 'individualsByGuid', 'variantTagsByGuid', 'familiesByGuid'])
+
+    @mock.patch('seqr.views.apis.staff_api.file_iter')
+    def test_upload_qc_pipeline_output(self, mock_file_iter):
+        url = reverse(upload_qc_pipeline_output,)
+        _check_login(self, url)
+
+        # Test no dataset type error
+        mock_file_iter.return_value = SAMPLE_QC_DATA_NO_DATA_TYPE
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'file': 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v15/sample_qc/final_output/seqr_sample_qc.tsv'}))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.reason_phrase, 'No dataset type detected')
+
+        # Test multiple dataset types error
+        mock_file_iter.return_value = SAMPLE_QC_DATA_MORE_DATA_TYPE
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'file': 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v15/sample_qc/final_output/seqr_sample_qc.tsv'}))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.reason_phrase, 'Multiple dataset types detected: wes ,wgs')
+
+        # Test unexpected data type error
+        mock_file_iter.return_value = SAMPLE_QC_DATA_UNEXPECTED_DATA_TYPE
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'file': 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v15/sample_qc/final_output/seqr_sample_qc.tsv'}))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.reason_phrase, 'Unexpected dataset type detected: "unknown" (should be "exome" or "genome")')
+
+        # Test normal functions
+        mock_file_iter.return_value = SAMPLE_QC_DATA
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'file': 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v15/sample_qc/final_output/seqr_sample_qc.tsv'}))
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['info', 'errors', 'warnings'])
+        self.assertListEqual(response_json['info'], [
+            u'Parsed 6 exome samples',
+            u'Found and updated matching seqr individuals for 4 samples'
+        ])
+        self.assertListEqual(response_json['warnings'], [
+            u'The following 2 samples were skipped: MANZ_1169_DNA, NA',
+            u'The following filter flags have no known corresponding value and were not saved: not_real_flag, contamination',
```

Comment:
contamination is a real flag, it corresponds to the PCT_CONTAMINATION column

Reply:
Ok, I will add the contamination flag back.